src/components/TDViewer/Property.jsx

```diff
@@ -0,0 +1,113 @@
+/********************************************************************************
+ * Copyright (c) 2018 - 2020 Contributors to the Eclipse Foundation
+ * 
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+import React, { useContext } from "react";
+import Swal from "sweetalert2";
+import "../../assets/main.css"
+import ediTDorContext from "../../context/ediTDorContext";
+import { buildAttributeListObject, separateForms } from "../../util.js"
+import addPropertyForm from "./AddPropertyForm";
+import Form from "./Form";
+
+const alreadyRenderedKeys = ["title", "forms", "description"];
+
+export default function Property(props) {
+    const context = useContext(ediTDorContext);
+
+    if ((Object.keys(props.prop).length === 0 && props.prop.constructor !== Object)) {
+        return <div className="text-3xl text-white">Property could not be rendered because mandatory fields are missing.</div>
+    }
+
+    const property = props.prop;
+    const forms = separateForms(props.prop.forms);
+
+    const attributeListObject = buildAttributeListObject({ name: props.propName }, props.prop, alreadyRenderedKeys);
+    const attributes = Object.keys(attributeListObject).map(x => {
+        return <li key={x}>{x} : {JSON.stringify(attributeListObject[x])}</li>
+    });
+
+    const checkIfFormExists = (form) => {
```

Comment:
This function seems to exists three times, once for each of Form (Property, Action, Event). Maybe reduce that to one and import it in the needed files.

Reply:
Yes we could do that. 
I will do it when I build the top level Forms